### ISSUE TITLE
Change OptionalUnit to report Unit returned by method implementations in interfaces

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt#3099]
## Expected Behavior of the rule
This gets flagged by OptionalUnit:

```kotlin
interface Foo {
  fun onMapClicked(point: Point?) = Unit
}
```

Right now though that use of Unit isn't reported.

## Context
This could instead be written as:

```kotlin
interface Foo {
  fun onMapClicked(point: Point?) {}
}
```

The use of Unit in the first example is therefore optional and should be reported. This means reverting the logic introduced in #1176.

**Comment by BraisGabin** [comment]
I think that this is a legit use of Unit. I think that this is better than an empty block. It demostrate more intentionality.

I'm not sure if I prefer that or an empty block with a `no-op`comment... But, any way, I don't think that this rule should flag that code.

**Comment by schalkms** [comment]
Brais is right. I remember this discussion from another thread. I think we decided to keep it like that, because otherwise the EmptyFunction rule would be triggered.

**Comment by 3flex** [comment]
> otherwise the EmptyFunction rule would be triggered

EmptyFunction ignores that case:
```kotlin
    override fun visitNamedFunction(function: KtNamedFunction) {
        super.visitNamedFunction(function)
        if (function.isOpen() || function.isDefaultFunction()) {
            return
        }
    ...
    }

    private fun KtNamedFunction.isDefaultFunction() =
        getParentOfType<KtClass>(true)?.isInterface() == true && hasBody()
```

So writing this should not flag any rule:
```kotlin
interface Foo {
  fun onMapClicked(point: Point?) {}
}
```

**Comment by arturbosch** [comment]
It was an intentional commit back then (7797809ad1c71f026a26ccbef425065484ebc931).
I'm fine with providing a stricter version via a config property.

**Comment by schalkms** [comment]
> EmptyFunction ignores that case:

It seems like that I missed this. This change came in after the linked PR.
Then I’m fine with changing the behavior back to the old one. I don’t think there’s a need for a config property here. Please go ahead.
